go.mod

```diff
@@ -20,6 +20,7 @@ require (
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 	gopkg.in/robfig/cron.v2 v2.0.0-20150107220207-be2e0b0deed5
 	gopkg.in/satori/go.uuid.v1 v1.2.0
+	gopkg.in/yaml.v3 v3.0.0-20191120175047-4206685974f2
```

Comment:
Consider using "sigs.k8s.io/yaml" instead.

Reply:
@nabokihms 
How would I approach parsing multi-document YAML with a v2 library? The streaming `Decoder` interface is not supported there.

Reply:
Something like this
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/yaml/decoder.go

I think the only way, for now, is to use this code from the apimachinery library.

Reply:
On the other hand, we can consider not using `sigs.k8s.io/yaml` :)

Reply:
> we can consider not using `sigs.k8s.io/yaml`

You've referenced a fundamental difference between v2 (sigs) and v3 versions of the package. v3 package would fail in the following case:

```yaml
a: a
a: b
```

Are we dependent on this?

Reply:
At the start, there was a ghodss/yaml, that was also used for addon-operator values. One moment I realize that the sigs.k8s.io/yaml is a successor of ghodss/yaml and it is a dependency of the client-go package.

Right now "sigs.k8s.io/yaml" is used to parse the hook configuration. This package can load "json-compatible" yaml (think map[string]interface{} instead of map[interface{}]interface{}) using json tags on structures. As JSON is a subset of YAML, the config is loaded directly as YAML https://github.com/flant/shell-operator/pull/239#discussion_r581820165.

At this time only JSON responses are supported from hooks. `NewDecoder` in json/encoding is used to parse "multi-document" or "streamed" JSON to support, for example, multiple metrics.

The feature to support multi-document YAML responses from hooks is a change of the protocol between shell-operator and a hook, that is why I'm asking "is it really needed?" [here](https://github.com/flant/shell-operator/pull/239#discussion_r579793228).

I think we should not change the protocol for one binging, but change it for all responses at once: metrics, conversion, validating.

So, fo me it looks like a separate PR where we can also migrate a config parser to yaml.v3.

Reply:
https://github.com/flant/shell-operator/issues/256